include/swift/Frontend/ArgsToFrontendOutputsConverter.h

```diff
@@ -41,7 +42,8 @@ class ArgsToFrontendOutputsConverter {
       : Args(args), ModuleName(moduleName), InputsAndOutputs(inputsAndOutputs),
         Diags(diags) {}
 
-  Optional<std::vector<std::string>> convert();
+  Optional<std::pair<std::vector<std::string>, SupplementaryOutputPaths>>
```

Comment:
I love the functional style, but since this will get even more complicated when a vector of SupplementaryPaths is returned, I'm changing it as suggested.

Reply:
I guess another alternative would be to use a one-off struct.

Reply:
Yes, that would work, too. I just went with out-parameters.